github3/models.py

```diff
@@ -42,6 +42,7 @@ def __init__(self, json, session=None):
 
         # set a sane default
         self._github_url = 'https://api.github.com'
+        self._ratelimit_resource = 'core'
```

Comment:
Could this be a class attribute, e.g.,

```py
class GitHubCore(...):
    _ratelimit_resource = 'core'
```

Then we could have

```
class GitHubSearchBase(GitHubCore):
    """..."""

    _ratelimit_resource = 'search'
```

without having to override `__init__`.

Reply:
[Done](https://github.com/sigmavirus24/github3.py/pull/776/commits/89ac06d18b87940d529aeac4a12f99c84c423fb0)